lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -0,0 +1,199 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { ApiFile } from '@cumulus/types/api/files';
+import {
+  CollectionPgModel,
+  GranulePgModel,
+  FilePgModel,
+  PostgresFile,
+  PostgresGranuleRecord,
+  translateApiGranuleToPostgresGranule,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from '@cumulus/errors';
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/granules' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+const { getBucket, getKey } = require('@cumulus/api/lib/FileUtils');
+const { deconstructCollectionId } = require('@cumulus/api/lib/utils');
+
+export interface GranulesAndFilesMigrationSummary {
+  granulesSummary: MigrationSummary,
+  filesSummary: MigrationSummary,
+}
+
+/**
+ * Migrate granules record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} record
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<any>}
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migrateGranuleRecord = async (
+  record: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<number> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(record, schemas.granule);
+  const { name, version } = deconstructCollectionId(record.collectionId);
+  const collectionPgModel = new CollectionPgModel();
+  const granulePgModel = new GranulePgModel();
+
+  const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+    knex,
+    { name, version }
+  );
+
+  const existingRecord = await knex<PostgresGranuleRecord>('granules')
+    .where({ granule_id: record.granuleId, collection_cumulus_id: collectionCumulusId })
+    .first();
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Granule ${record.granuleId} was already migrated, skipping`);
+  }
+
+  const granule = await translateApiGranuleToPostgresGranule(record, knex, collectionPgModel);
+  const [cumulusId] = await granulePgModel.upsert(knex, granule);
+  return cumulusId;
+};
+
+/**
+ * Migrate File record from a Granules record from DynamoDB  to RDS.
+ *
+ * @param {ApiFile} file - Granule file
+ * @param {number} granuleCumulusId - ID of granule
+ * @param {Knex} knex - Knex client for writing to RDS database
```

Comment:
In reality this parameter is now a Knex transaction, so we should update the JSdoc

Reply:
[`bc6cc2b` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/bc6cc2bd0d21127123f6a3bbe6cc5477bd3579fc) Don't I need to update the type too?

Reply:
Yup